### PR TITLE
Improve compatibility to uglyurls

### PR DESCRIPTION
**Pull request by dakatsuka**
When Hugo's uglyurls is enabled, pagination and taxonomy urls are broken, so I fixed it. In the current version of Hugo can't get uglyurls option in layouts, so we need to add it to the parameter 🙏 

Thank you for creating great theme!

### CODE REVIEW

File: layouts/_default/list.html [hunk 1]
```diff
@@ -15,16 +15,28 @@
       <ul class="pager">
         {{ if .Paginator.HasPrev }}
           <li class="previous">
+            {{ if $.Site.Params.uglyurls }}
+            <a href="{{ replaceRE "\\.html$" "/" .Permalink }}page/{{ .Paginator.Prev.PageNumber }}.html"
```
Comment (dakatsuka) [inline layouts/_default/list.html:19]: 📝 If you don't use replaceRE, URL will be broken. It looks: `exmaple.com/tags/tagname.htmlpage/2.html`

**Comment by matsuyoshi30** [comment]
@dakatsuka Thanks for your contribution! I'll take a look later.

File: README.md [hunk 2]
```diff
@@ -221,4 +232,4 @@ $ npm run build-prod
 [MIT](./LICENSE).
 
 
-[![FOSSA Status](https://app.fossa.com/api/projects/git%2Bgithub.com%2Fmatsuyoshi30%2Fharbor.svg?type=large)](https://app.fossa.com/projects/git%2Bgithub.com%2Fmatsuyoshi30%2Fharbor?ref=badge_large)
\ No newline at end of file
+[![FOSSA Status](https://app.fossa.com/api/projects/git%2Bgithub.com%2Fmatsuyoshi30%2Fharbor.svg?type=large)](https://app.fossa.com/projects/git%2Bgithub.com%2Fmatsuyoshi30%2Fharbor?ref=badge_large)
```
Comment (matsuyoshi30) [inline README.md:235]: I wonder why this line is out as different...

File: layouts/_default/list.html (hunk 1 above)
Comment (matsuyoshi30) [reply layouts/_default/list.html:19]: Great infomation. Thanks!

**Comment by matsuyoshi30** [comment]
LGTM! Thanks for removing unnecessary slash in tags and categories too!

I wonder why FOSSA line is out as different in README. Could you fix it?

File: README.md (hunk 2 above)
Comment (dakatsuka) [reply README.md]: Original README.md did not seem to have a line feed at the end of file. vim i used was set to automatically insert newline at the end of file, which seems to have caused the difference 🙇

Comment (matsuyoshi30) [reply README.md]: Thanks for the quick response!
I think there should be a newline at the end of the file... Sorry, I'll fix it later after merge it.

**Comment by matsuyoshi30** [comment]
@dakatsuka Thanks for your contribution!

**Pull request merged**